DirectX11/HackerDXGI.cpp

```diff
@@ -167,6 +167,104 @@ HackerDXGISwapChain::HackerDXGISwapChain(IDXGISwapChain *pSwapChain, HackerDevic
+{
+	mOrigSwapChain = pSwapChain;
+
+	if (pContext == NULL)
+	{
+		pDevice->GetImmediateContext(reinterpret_cast<ID3D11DeviceContext**>(&pContext));
+	}
+
+	mHackerDevice = pDevice;
+	mHackerContext = pContext;
+	pDevice->SetHackerSwapChain(this);
+
+	// TODO: multisampled swap chain
+	// TODO: swap chain with more than one back buffer
+	D3D11_TEXTURE2D_DESC fake_buffer_desc;
+	std::memset(&fake_buffer_desc, 0, sizeof(D3D11_TEXTURE2D_DESC));
+	fake_buffer_desc.ArraySize = 1;
+	fake_buffer_desc.MipLevels = 1;
+	fake_buffer_desc.BindFlags = D3D11_BIND_RENDER_TARGET | D3D11_BIND_SHADER_RESOURCE;
+	fake_buffer_desc.Usage = D3D11_USAGE_DEFAULT;
+	fake_buffer_desc.SampleDesc.Count = 1;
+	fake_buffer_desc.Format = fake_swap_chain_desc->BufferDesc.Format;
+	fake_buffer_desc.MiscFlags = 0;
+	fake_buffer_desc.Width = fake_swap_chain_desc->BufferDesc.Width;
+	fake_buffer_desc.Height = fake_swap_chain_desc->BufferDesc.Height;
+	fake_buffer_desc.CPUAccessFlags = 0;
+
+	HRESULT res = pDevice->CreateTexture2D(&fake_buffer_desc, nullptr, &mFakeBackBuffer);
+
+	mWidth = new_width;
+	mHeight = new_height;
+
+	if (res != S_OK)
+	{
+		LogInfo(" *** Failed to create faked swap chain buffer! \n");
+		mFakeBackBuffer = nullptr;
+		throw;
+	}
+
+	try {
+		// Create Overlay class that will be responsible for drawing any text
+		// info over the game. Using the Hacker Device and Context we gave the game.
+		mOverlay = new Overlay(pDevice, pContext, this);
+	}
+	catch (...) {
+		LogInfo("  *** Failed to create Overlay. Exception caught. \n");
+		mOverlay = NULL;
+	}
+}
+
+HackerDXGISwapChain::HackerDXGISwapChain(IDXGISwapChain *pSwapChain, HackerDevice *pDevice, HackerContext *pContext, IDXGISwapChain* fake_swap_chain, UINT new_width, UINT new_height)
+	: HackerDXGIDeviceSubObject(pSwapChain), mFakeBackBuffer(nullptr), mFakeSwapChain(nullptr), mWidth(0), mHeight(0)
+{
+
+	mOrigSwapChain = pSwapChain;
+
+	if (pContext == NULL)
+	{
+		pDevice->GetImmediateContext(reinterpret_cast<ID3D11DeviceContext**>(&pContext));
+	}
+
+	mHackerDevice = pDevice;
+	mHackerContext = pContext;
+	pDevice->SetHackerSwapChain(this);
+
+	mFakeSwapChain = fake_swap_chain;
+
+	if (mFakeSwapChain == nullptr)
+	{
+		LogInfo(" *** Failed to create hacker swap chain: provided faked swap chain is invalid! \n");
+		throw;
+	}
+
+	mWidth = new_width;
+	mHeight = new_height;
+
+	try {
+		// Create Overlay class that will be responsible for drawing any text
+		// info over the game. Using the Hacker Device and Context we gave the game.
+		mOverlay = new Overlay(pDevice, pContext, this);
+	}
+	catch (...) {
+		LogInfo("  *** Failed to create Overlay. Exception caught. \n");
+		mOverlay = NULL;
+	}
+}
+
+HackerDXGISwapChain::~HackerDXGISwapChain()
+{
+	if (mFakeSwapChain)
+		mFakeSwapChain->Release();
+	if (mFakeBackBuffer)
+		mFakeBackBuffer->Release();
+}
+
+
 HackerDXGISwapChain1::HackerDXGISwapChain1(IDXGISwapChain1 *pSwapChain1, HackerDevice *pDevice, HackerContext *pContext)
```

Comment:
Yeah, i was gonna ask you, whether the other paths are used at all. All my games that i have tested use normal swapchain object. With the time  Ill add the upscaling code in those functions too. Maybe you can list some games besides dishonored that are using another swpchain creation path, so i can test the updates directly.